src/router.js

```diff
@@ -35,6 +36,35 @@ const logoutAndRedirect = async (to, from, next) => {
   return next('/login');
 };
 
+const checkAndSubscribeToLobby = async (to, from, next) => {
+  const gameStore = useGameStore();
+  const authStore = useAuthStore();
+  const gameId = parseInt(to.params.gameId);
+  
+  if (Number.isNaN(gameId) || !Number.isFinite(gameId)) {
+    to.meta.error = 'Invalid lobby number';
+    return next();
+  }
+
+  if (!authStore.authenticated) {
+    return next(`/login/${gameId}`);
+  }
+
+  if (gameStore.players.some(({username}) => username === authStore.username)) {
+    return next();
+  }
+  
+  try {
+    await gameStore.requestSubscribe(+gameId);
+    next();
+  }
+  catch (err) {
+    to.meta.error = err.message;
+    next();
+  }
```

Comment:
How does this affect the flow when a user hits the join game button on a lobby and the request fails because the game is full? Does the user still get a snackbar on the home view or are they now redirected to the lobby error view?

Reply:
The snackbar still shows up, the router does not push the lobby url unless the `requestSubscribe` in successful, otherwise it emits the error which displays the snackbar. I confirmed this with cypress just to be sure